WordPress/gutenberg
# FormTokenField: add ability to auto select first matching suggestion for incomplete token

## What?

This PR adds a new experimental prop to `<FormTokenField>` allows users to select the first matching suggestion for an incomplete token by pressing the Enter key. This is meant to use in tandem with `__experimentalValidateInput`.
## Why?


This is useful when `<FormTokenField>` is used as a dropdown selector. Users don't need to type the full value of the suggestion to select it with the Enter key.

For example, given a component with `['blue', 'red', 'green']` as the suggestion array, users can select `blue` by typing `b` or `bl` only and then press Enter.

## How?


## Testing Instructions


1. In the storybook, navigate to FormTokenField > Dropdown Selector story.
2. Type `an` then press Enter key.
3. `Antarctica` should be selected.

## Screenshots or screencast 

~https://user-images.githubusercontent.com/5423135/179729142-c424170b-4eaf-4904-931c-2687633f1413.mov~

Updated screencast

https://user-images.githubusercontent.com/5423135/180684293-eaa2105a-9ecd-4fa0-b8c3-6071c025cd6e.mov

## Changelog

FormTokenField: add `__experimentalAutoSelectFirstMatch` prop to auto select the first matching suggestion on typing.
